### ISSUE TITLE
Can't get badge count working in background mode

### ISSUE THREAD

**Issue by patriciy** [peter-iakovlev/Telegram-iOS#30]
Assembling Fork configuration.
Push messages seems to work fine.
But in background mode I don't see badge counter on app icon is being updated.
Stock telegram seems fine.
Any special configuration needed for this?

**Comment by shivanraptor** [comment]
Did you change the Bundle Identifier?

**Comment by patriciy** [comment]
@shivanraptor Yes, I changed Bundle ID and group ID. No changes with related code too.

**Comment by shivanraptor** [comment]
Try to select Settings > Notifications and Sounds > Reset All Notifications and retry.

**Comment by shivanraptor** [comment]
See #34

**Comment by patriciy** [comment]
That's all reseted, no luck

**Comment by shivanraptor** [comment]
Problem still exists in release-5.6. No badge icon count in background mode. When comparing side-by-side with official Telegram App, the badge icon count is working in the official one.

**Comment by patriciy** [comment]
I investigated this a lot. I checked stock Telegram's app logs side by side with my custom app.
I think all magic happening on VoIP push received.
And when VoIP push token is sent to telegram server that's what I receive
`response for 6695690508489876480 is error: 400: APP_PUSH_CERT_MISSING]`

So basically I need to upload VoIP certificate to Telegram server. 
But I can't find any options to do this.

**Comment by patriciy** [comment]
#49
